compiler-core/src/build/project_compiler.rs

```diff
@@ -328,6 +329,156 @@ where
+            tracing::info!("skipping_mix_build_for_non_erlang_target");
+            return Ok(());
+        }
+
+        let build_dir = paths::build_packages(mode, target);
+        let project_dir = paths::build_deps_package(name);
+        let mix_build_dir = project_dir.join("_build").join(mix_target);
+        // Absolute build path is needed for mix to make accurate symlinks
+        let mix_build_path = std::env::current_dir()
+            .expect("Project root")
+            .join(&mix_build_dir);
+        let mix_build_lib_dir = mix_build_dir.join("lib");
+        let up = paths::unnest(&project_dir);
+        let mix_path = |path: &Path| up.join(path).to_str().unwrap_or_default().to_string();
+        let ebins = paths::build_packages_ebins_glob(mode, target);
+        let dest = paths::build_package(mode, target, name);
+
+        // Elixir core libs must be loaded
+        self.maybe_link_elixir_libs(&build_dir)?;
+
+        // Prevent Mix.Compilers.ApplicationTracer warnings
+        // mix would make this if it didn't exist, but we make it anyway as
+        // we need to link the compiled dependencies into there
+        self.io.mkdir(&mix_build_lib_dir)?;
+        let deps = &package.requirements;
+        for dep in deps {
+            // TODO: unit test
+            let dep_source = build_dir.join(&dep);
+            let dep_dest = mix_build_lib_dir.join(&dep);
+            if self.io.is_directory(&dep_source) && !self.io.is_directory(&dep_dest) {
+                tracing::debug!("linking_{}_to_build", dep);
+                self.io.symlink_dir(&dep_source, &dep_dest)?;
+            }
+        }
+
+        let env = [
+            ("MIX_BUILD_PATH", mix_path(&mix_build_path)),
+            ("MIX_ENV", mix_target.into()),
+            ("MIX_QUIET", "1".into()),
+            ("TERM", "dumb".into()),
+        ];
+        let args = [
+            "-pa".into(),
+            mix_path(&ebins),
+            "-S".into(),
+            "mix".into(),
+            "compile".into(),
+            "--no-deps-check".into(),
+            "--no-load-deps".into(),
+            "--no-protocol-consolidation".into(),
+        ];
+        let status = self.io.exec(
+            "elixir",
+            &args,
+            &env,
+            Some(&project_dir),
+            self.silence_subprocess_stdout,
+        )?;
+
+        if status == 0 {
+            // TODO: unit test
+            let source = mix_build_dir.join("lib").join(name);
+            if self.io.is_directory(&source) && !self.io.is_directory(&dest) {
+                tracing::debug!("linking_{}_to_build", name);
+                self.io.symlink_dir(&source, &dest)?;
+            }
+            Ok(())
+        } else {
+            Err(Error::ShellCommand {
+                program: "mix".to_string(),
+                err: None,
+            })
+        }
+    }
+
+    fn maybe_link_elixir_libs(&mut self, build_dir: &PathBuf) -> Result<(), Error> {
```

Comment:
Could we get some comments in this function explaining what it does at each step and why we need to do it? To make it easier for newcomers and also myself in a couple years time!